src/screens/Product/Product.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import { Text, View, ScrollView, Alert, BackHandler } from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import CustomButton from 'src/components/CustomButton';
+import styles from './styles';
+
+class Product extends Component {
+  static navigationOptions = ({ navigation }) => ({
+    headerLeft: null,
+    header: () => (<View style={styles.headerWrapper}>
+      <Icon size={50} name={navigation.getParam('iconName')} />
+      <Text style={styles.headerText}>{navigation.getParam('title')}</Text>
+    </View>)
+  });
+
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
+  }
+
+  handleBackPress = () => {
+    Alert.alert(
```

Comment:
It's good that you tried on this API but in mobile application it's not a good UX to block transition with confirm alerts. The latest navigation has already provided `hardwareBackPress` logic, so remove please redundant listener.